docs/userguide.rst

```diff
@@ -315,6 +315,33 @@ and push the created image to the DockerHub image registry:
 
     $ docker push johndoe/myenv
 
+**Supporting non-root users**
+
+Running containers with root user is not secure and therefore dicouraged. If
+you want to improve the security in your environment you can setup your own
+user.
+
+In order to be compatible with the REANA platform you should do the following:
```

Comment:
In order for processesqo to run under any user identities and still be able to write to shared workspaces, we use a GID=0 technique as used by OpenShift  https://docs.openshift.com/container-platform/3.11/creating_images/guidelines.html#openshift-specific-guidelines  :

- UID: you can use any user ID you want;
- GID: your should add your user to group with GID=0 (the `root` group)

This will ensure the writable access to workspace directories managed by the REANA platform.

Fer example, you can create the user `johndoe` with UID=501 and add the user to GID=0 by following:

...